server/mscalendar/filters.go

```diff
@@ -31,6 +31,13 @@ func withRemoteUser(user *User) func(m *mscalendar) error {
 	}
 }
 
+func withActingUser(mattermostUserID string) func(m *mscalendar) error {
```

Comment:
Not cheating per se, but maybe error-prone. This is being called in a service account context, so we are changing the state of `m` temporarily while handling a given user's events. This makes it difficult to do concurrent operations across users.

Reply:
This is the only way I imagined of leaving the architecture as it is, without moving other parts of the logic outside of the main "calendar" logic, but I do not know how this could affect elsewhere, if anything. Only running under GCal at the moment. Open to suggestions!

Reply:
In general this seems fine, though just concerned about potential pitfalls of mutating the state. Really that's just an artifact of the `.Filter()` method in general. It's mainly meant to be used in a per-user context, but we introduced the superuser token later. And now we're doing sort of a hybrid, where we're using user-level OAuth tokens in a scheduled job context.

I think it's fine how this PR implements it, though we will need to rethink this if we want to perform these requests concurrently across different users. Speaking of which, what are your thoughts on the performance of the current approach, versus fanning out the requests in some fashion? The timing of reminders etc. depends on the speed of this operation, so I'm thinking we'll need some performance optimization here in the case of sending individual requests for each user.

Reply:
In order to parallelize work we can do a `Filter` function that returns a copy of the `mscalendar` with the context switched so we can send out all the jobs to goroutines and wait for all of them to finish using a waitgroup. Is not the best approach, but it surely will help.